CMakeLists.txt

```diff
@@ -438,7 +448,7 @@ endif()
 
 set(USE_X11 0)
 
-if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT ENABLE_HEADLESS)
+if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT HAIKU AND NOT ENABLE_HEADLESS)
```

Comment:
I disagree with that. If you know a clean way to fix something, just do it then!